### PR TITLE
LPD-26289

### PR DESCRIPTION
**Pull request by lucasmiranda0**
Related issue [LPD-26289](https://liferay.atlassian.net/browse/LPD-26289)
Changes asked here: https://github.com/natocesarrego/liferay-docker/pull/54

### CODE REVIEW

File: release/_bom.sh [hunk 1]
```diff
@@ -224,58 +224,46 @@ function generate_pom_release_dxp_bom_third_party {
 		-e "w ${pom_file_name}" \
 		"${_RELEASE_TOOL_DIR}/templates/release.dxp.bom.third.party.pom.tpl" > /dev/null
 
-	local property_files=(
-		"${_PROJECTS_DIR}/liferay-portal-ee/lib/development/dependencies.properties"
-		"${_PROJECTS_DIR}/liferay-portal-ee/lib/portal/dependencies.properties"
-	)
+	IFS=$'\n' read -d '' -ra dependencies_properties < "${_PROJECTS_DIR}/liferay-portal-ee/lib/development/dependencies.properties"
+	IFS=$'\n' read -d '' -ra portal_dependencies_properties < "${_PROJECTS_DIR}/liferay-portal-ee/lib/portal/dependencies.properties"
```
Comment (natocesarrego) [inline release/_bom.sh:228]: @lucasmiranda0, use the following isn't enough?
```
IFS=$'\n' read -d '' -ra dependencies_properties << "${_PROJECTS_DIR}/liferay-portal-ee/lib/portal/dependencies.properties"
```
I mean, concatenate the content of `${_PROJECTS_DIR}/liferay-portal-ee/lib/portal/dependencies.properties` into the existing array named `dependencies_properties`

File: release/_bom.sh [hunk 2]
```diff
@@ -224,58 +224,46 @@ function generate_pom_release_dxp_bom_third_party {
...
-					do
-						echo -e "\t\t\t<dependency>"
-						echo -e "\t\t\t\t<groupId>${group_id}</groupId>"
-						echo -e "\t\t\t\t<artifactId>${artifact_id}</artifactId>"
-						echo -e "\t\t\t\t<version>${version}</version>"
-						echo -e "\t\t\t</dependency>"
-					done >> "${pom_file_name}"
-			fi
-		done
+	local compile_only_file_name="release.dxp.bom.compile.only-${_PRODUCT_VERSION}-${_BUILD_TIMESTAMP}.pom"
```
Comment (natocesarrego) [inline release/_bom.sh:232]: @lucasmiranda0, can we rename to `pom_compile_only_file_name`?

File: release/_bom.sh [hunk 3]
```diff
@@ -224,58 +224,46 @@ function generate_pom_release_dxp_bom_third_party {
...
+
+	local dependencies_included=()
+
+	for dependency_property in "${dependencies_properties[@]}"
+	do
+		IFS=':' read -ra dependency_property_parts <<< "$(echo "${dependency_property}" | cut -d= -f2)"
+
+		if [[ ${dependencies_included[@]} =~ "${dependency_property_parts[0]}${dependency_property_parts[1]}${dependency_property_parts[2]}" ]]
+		then
+			echo "${dependency_property_parts[0]}${dependency_property_parts[1]}"
```
Comment (natocesarrego) [inline release/_bom.sh:242]: @lucasmiranda0, is it really necessary?

File: release/_bom.sh [hunk 4]
```diff
@@ -224,58 +224,46 @@ function generate_pom_release_dxp_bom_third_party {
...
-					done >> "${pom_file_name}"
-			fi
-		done
+	local compile_only_file_name="release.dxp.bom.compile.only-${_PRODUCT_VERSION}-${_BUILD_TIMESTAMP}.pom"
+
+	local dependencies_included=()
+
+	for dependency_property in "${dependencies_properties[@]}"
+	do
+		IFS=':' read -ra dependency_property_parts <<< "$(echo "${dependency_property}" | cut -d= -f2)"
```
Comment (natocesarrego) [inline release/_bom.sh:238]: @lucasmiranda0, can you add blank spaces in the `cut` command parameters?

`cut -d = -f 2`

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: release/_bom.sh (hunk 1 above)
Comment (lucasmiranda0) [reply release/_bom.sh:228]: I tried that but it didn't work.

File: release/_bom.sh (hunk 3 above)
Comment (lucasmiranda0) [reply release/_bom.sh:242]: No, sorry for that.

**Pull request closed without merging**